calcom/cal.com
# test: e2e oidc

## What does this PR do?
E2E Test for OIDC. This end to end test has been made in reference to the docs [here](https://cal.com/docs/how-to-guides/how-to-setup-oidc-with-okta). Kindly follow the steps here to first get the .env

Fixes: #11295 

## Requirement/Documentation
Setup these variables in the .env 
```
# This user will be the SAML ADMIN who will have the admin rights.
# Make sure the email is in the SAML_ADMINS list as well 
E2E_TEST_SAML_ADMIN_EMAIL=
E2E_TEST_SAML_ADMIN_PASSWORD=

# Get this information from an OIDC Provider like Okta
E2E_TEST_OIDC_CLIENT_ID=
E2E_TEST_OIDC_CLIENT_SECRET=
# Domain Example "dev-83149661-admin.okta.com"
E2E_TEST_OIDC_PROVIDER_DOMAIN=

# Add an email and password in the OIDC Provider (Creating a User in `People` Section in Okta) first, then add them here. In 
E2E_TEST_OIDC_USER_EMAIL=
E2E_TEST_OIDC_USER_PASSWORD=
```

## Type of change



- [x] Chore/Test (refactoring code, technical debt, workflow improvements)

## How should this be tested?
- Make sure the above .env variables are set.
- Ensure the SAML_ADMIN user doesn't already exist in the DB
- Run `yarn dx` or `yarn dev`
- Run `yarn e2e oidc` to run the test.
